File: real-estate-bidding-system-api/src/main/java/org/universe/realestatebiddingsystem/user/model/response/UserResponseModel.java
package org.universe.realestatebiddingsystem.user.model.response;

import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class UserResponseModel {
    private Long id;
    private String email;
    private String firstName;
    private String lastName;
    private String telephone;
    private String town;
    private List<RoleResponseModel> rolesList = new ArrayList<>();
    private Boolean isBanned = false;
}

File: real-estate-bidding-system-client/src/app/core/model/response/user-response.model.ts
import {RoleResponseModel} from './role-response.model';

export class UserResponseModel {
  constructor(
    public id: string,
    public email: string,
    public firstName: string,
    public lastName: string,
    public telephone: string,
    public town: string,
    public isBanned: boolean,
    public rolesList: RoleResponseModel[]
  ) {
  }
}
